src/deepMap.ts

```diff
@@ -73,18 +93,23 @@ export class DeepMapEntry<T> {
  * @private
  */
 export class DeepMap<T> {
```

Comment:
Please resolve if out of scope!! 🙏

Reply:
Yep it could definitely be considered, I wasn't sure of the reason for the current API. I kept this change to the same API for now so I could write tests that showed the new impl was equivalent